test/Node_test.lua

```diff
@@ -0,0 +1,56 @@
+luaunit = require("luaunit")
+Node = require("Node")
+
+local TestNode = {}
+
+local root, node_1, node_2, node_3 = Node:new(), Node:new(), Node:new(), Node:new()
+root:add_child(node_1)
+root:add_child(node_2)
+node_2:add_child(node_3)
+
+function TestNode:test_new()
+  luaunit.assertIs(getmetatable(root), Node)
+
+  luaunit.assertNotNil(root)
+
+  luaunit.assertNil(root.parent)
```

Comment:
When I add this assertion, the test fails:

Failed tests:
-------------
1) Test_Node.test_new
.\test\Node_test.lua:18: Contents of the tables are not identical:
Expected:
{}
Actual:
<table: 00630460> {
    <table: 00630488> {
        children=<table: 006304B0> {},
        parent=<table: 00630438> {children=<table: 00630460>}
    },
    <table: 006304D8> {
        children=<table: 00630500> {<table: 00630528> {children=<table: 00630550> {}, parent=<table: 006304D8>}},
        parent=<table: 00630438>
    }
}
stack traceback:
        .\test\Node_test.lua:18: in function 'Test_Node.test_new'